core/src/main/java/org/kiwix/kiwixmobile/core/main/CoreMainActivity.kt

```diff
@@ -109,6 +110,15 @@ abstract class CoreMainActivity : BaseActivity(), WebViewProvider {
     if (destination.id !in topLevelDestinations) {
       handleDrawerOnNavigation()
     }
+    if (destination.id == readerFragmentResId) {
```

Comment:
when you are in a scenario where both parts of an if statement are identical except for some parameters then it is a hint you are using the if too broadly eg
```
if(condition)
aMethod(1)
else
aMethod(2)
```
can always be rewritten as

```
aMethod(if(condition) 1 else 2)
```

and now what the code is saying a lot more clearly is that the condition is controlling the input to the method.

You may want to add imports to avoid declaring `DrawerLayout.` to save on space so fits on 1 line

Reply:
@macgills Thank you!